File: app/schemas/images.py
"""06 图片库：ImageResponse（蓝图 06 schema 段）。"""

import uuid
from datetime import datetime

from pydantic import BaseModel


class ImageResponse(BaseModel):
    """图片展示信息（url 为短时 token 展示端点，前端直接 <img>）。"""

    id: uuid.UUID
    url: str
    thumbnail_url: str | None = None
    filename: str
    content_type: str
    size: int
    created_at: datetime


class ImageSearchRequest(BaseModel):
    """文字搜图入参（multipart 图片搜图走 File）。"""

    query_text: str
    top_k: int = 10

File: frontend/src/api/images.ts
import { api } from './http'
import type { Page } from './chat'

export interface ImageItem {
  id: string
  url: string
  thumbnail_url?: string | null
  filename: string
  content_type: string
  size: number
  created_at: string
}

export const listImages = (page = 1, pageSize = 24) =>
  api<Page<ImageItem>>({ method: 'GET', url: '/images', params: { page, page_size: pageSize } })

export const uploadImage = (file: File) => {
  const form = new FormData()
  form.append('file', file)
  return api<ImageItem>({ method: 'POST', url: '/images', data: form })
}

export function searchImages(file?: File, queryText?: string) {
  const form = new FormData()
  if (file) form.append('file', file)
  if (queryText) form.append('query_text', queryText)
  return api<ImageItem[]>({ method: 'POST', url: '/images/search', data: form })
}

export const requestDeleteImage = (id: string) =>
  api<{ delete_token: string }>({ method: 'DELETE', url: `/images/${id}` })

export const confirmDeleteImage = (id: string, token: string) =>
  api<null>({ method: 'POST', url: `/images/${id}/confirm`, data: { delete_token: token } })
